src/compiler/scala/tools/nsc/transform/LazyVals.scala

```diff
@@ -54,14 +54,35 @@ abstract class LazyVals extends Transform with TypingTransformers with ast.TreeD
     private val lazyVals = perRunCaches.newMap[Symbol, Int]() withDefaultValue 0
 
     import symtab.Flags._
+    private def flattenThickets(stats: List[Tree]): List[Tree] = stats.flatMap(_ match {
```

Comment:
I think there may be more opportunities in MethodSynthesis, but we can probably just make a mental note and come back to this.